[Serializer] Add feature description for `AbstractNormalizer::REQUIRE_ALL_PROPERTIES` context option (#17979)
symfony/symfony-docs: components/serializer.rst

    $result = $normalizer->normalize($dummy, 'json', [AbstractObjectNormalizer::SKIP_NULL_VALUES => true]);
    // ['bar' => 'notNull']

> Preventing ``null`` value fallback for nullable properties

Reviewer: ```suggestion
Preventing ``null`` Value Fallback for Nullable Properties
```

Author: @OskarStark Aren't all nouns written in lowercase in English (except countries and languages)? Or is this a convention of the Symfony documentation?

Reviewer: For reference:
https://symfony.com/doc/current/contributing/documentation/standards.html#english-language-standards

Author: @OskarStark Thanks for the reference, I've committed the changes 🙂

Reviewer: Thank you!

----------------------------------------------------------

> By default, the Serializer will add `null` to nullable properties when the parameters for those are not provided.

Reviewer: ```suggestion
By default, the Serializer will add ``null`` to nullable properties when the parameters for those are not provided.
```

You can change this behavior by setting the ``AbstractNormalizer::PREVENT_NULLABLE_FALLBACK`` context option
to ``true``::

>     class Dummy {

Reviewer: ```suggestion
    class Dummy
    {
```

        public function __construct(
            public string $foo,
            public ?string $bar,
        ) {
        }
    }

    $data = ['foo' => 'notNull'];

    $normalizer = new ObjectNormalizer();
    $result = $normalizer->denormalize($data, Dummy::class, 'json', [AbstractNormalizer::PREVENT_NULLABLE_FALLBACK => true]);
    // throws Symfony\Component\Serializer\Exception\MissingConstructorArgumentException

.. versionadded:: 6.3

>     The context flag was introduced in Symfony 6.3.

Reviewer: ```suggestion
    The ``AbstractNormalizer::PREVENT_NULLABLE_FALLBACK`` context option
    was introduced in Symfony 6.3.
```
